token-metadata-creator/app/Config.hs

```diff
@@ -196,8 +196,35 @@ wellKnownOption
     => OA.Mod OA.OptionFields p
     -> OA.Parser p
 wellKnownOption =
-    OA.option wellKnownReader
+  OA.option (OA.eitherReader (\arg ->
+                               let
+                                 r1 = asJSONValue arg
+                               in
+                                 if isPotentiallyString r1
+                                 then asJSONString arg
+                                 else r1
+                             ))
   where
-    wellKnownReader :: OA.ReadM p
-    wellKnownReader = OA.eitherReader $
-        Aeson.parseEither parseWellKnown . Aeson.toJSON
+    -- Presume we got a value (i.e. 3 or []).
+    asJSONValue :: String -> Either String p
+    asJSONValue = (Aeson.parseEither parseWellKnown =<<) . Aeson.eitherDecodeStrict' . BC8.pack
+    -- Presume we got a string (i.e. "3" or "potato").
+    asJSONString :: String -> Either String p
+    asJSONString = Aeson.parseEither parseWellKnown . Aeson.String . T.pack
```

Comment:
```suggestion
    asJSONString :: ByteString -> Either String p
    asJSONString = (Aeson.parseEither parseWellKnown . bimap show Aeson.String) <=< T.decodeUtf8'
```
Use `Data.Text.Encoding.decodeUtf8'` to ensure that text is always treated as UTF-8.